### PR TITLE
feat(overlay): emit all build-mode output formats

### PR DESCRIPTION
**Pull request by yktyagi**
#### Merge Checklist  

**All** boxes should be checked before merging the PR

- [x] The changes in the PR have been built and tested
- [x] Documentation has been updated to reflect the changes (or no doc update needed)
- [x] Ready to merge

#### Description 

Overlay mode always emitted a `raw` artifact regardless of the template's
`disk.artifacts`, so requesting `qcow2` (or `vhd`/`vhdx`/`vmdk`/`vdi`) silently
produced only a `.raw` file — the emit stage hardcoded `<name>-<version>.raw`
and did no format conversion.

This adds a **`Convert Artifacts`** stage to the overlay `Postprocess` that runs
the same `imageconvert.ConvertImageFile` the create-mode `rawmaker` uses, keyed on
the template's `disk.artifacts`. As a result, overlay mode now supports every
output format and compression option that create/build mode does.

Behavior details:

- The stage runs **after** the RAW inspection, because a request that omits `raw`
  deletes the `.raw` file as part of conversion (mirroring create mode).
- It is a **no-op** when `disk.artifacts` is empty or lists only `raw`, so plain
  RAW overlay builds are unchanged.
- Wired behind a test seam (`builderConvertFn`) consistent with the other overlay
  stage seams; a conversion failure surfaces from `Postprocess` and the mount/loop
  cleanup chain still runs exactly once.

#### Any Newly Introduced Dependencies

None. Reuses the existing `internal/image/imageconvert` package (`qemu-img`), which
is already a build-mode dependency.

#### How Has This Been Tested? 

- `go build` / `go vet` on the affected packages (`internal/image/overlay`,
  `internal/image/imageconvert`, `internal/provider/...`, `cmd/...`) — clean.
- `go test ./internal/image/overlay/` — passes, including:
  - updated happy-path and inspect-order stage assertions to include the new
    `Convert Artifacts` stage;
  - new `TestBuilder_ConvertFailureFailsPostprocess` verifying a conversion
    failure surfaces and cleanup still runs once.
- Verified end-to-end against a real overlay build (`build --baseline-image ...
  noble-server-cloudimg-amd64-overlay.yml`) whose template declares
  `disk.artifacts: [{ type: qcow2 }]`; the build now emits the `qcow2` artifact.

### CODE REVIEW

File: internal/image/overlay/session_test.go [hunk 1]
```diff
@@ -188,7 +196,8 @@ func TestBuilder_HappyPathOrdersStagesAndCleansUp(t *testing.T) {
 		t.Fatalf("Postprocess: %v", err)
 	}
 
-	want := []string{"acquire", "mount", "detect", "resolve", "preflight", "resize", "install", "configure", "regenBoot", "grubRegen", "sbom", "emit:1.0"}
+	// Inspection is disabled by default, so conversion runs directly after emit.
+	want := []string{"acquire", "mount", "detect", "resolve", "preflight", "resize", "install", "configure", "regenBoot", "grubRegen", "sbom", "emit:1.0", "convert:/out/img-1.0.raw"}
```
[the remark this answers is not in the document]
Comment (yktyagi) [reply internal/image/overlay/session_test.go]: Good catch — reworded the comment. Inspection is skipped in this test only because the test template leaves `InspectEnabled` at its zero value (`false`); it's not the product default. The CLI defaults `--inspect` on, and that path is covered by `TestBuilder_InspectRunsAfterEmitWhenEnabled`, which the comment now points to.

File: internal/image/overlay/session.go [hunk 2]
```diff
@@ -415,6 +419,22 @@ func (b *Builder) Postprocess(buildErr error) (err error) {
...
+	// vhdx, vmdk, vdi) and apply any configured compression, so overlay mode emits
+	// every output format create/build mode supports. This reuses the same
+	// converter the create-mode rawmaker runs, keyed on template disk.artifacts;
+	// it is a no-op when disk.artifacts is empty or lists only raw, so plain RAW
+	// overlay builds are unchanged. It runs AFTER the RAW inspection above because
+	// a request that omits raw deletes the RAW file as part of conversion.
+	if err := b.timeStage("Convert Artifacts", func() error {
+		if cerr := builderConvertFn(artifact, b.template); cerr != nil {
+			return fmt.Errorf("overlay postprocess: failed to convert image artifact: %w", cerr)
+		}
```
[the remark this answers is not in the document]
Comment (yktyagi) [reply internal/image/overlay/session.go:432]: Fixed. The qemu-img `convert` invocations (and the `virt-sparsify`/sparsify fallbacks) now single-quote both the source and destination paths via the package's `shellSingleQuote` helper, and pass `--` before the positional paths — the same pattern `DetectImageFormat` already used. This neutralizes spaces and shell metacharacters in artifact paths, so the newly-reachable overlay conversion path (and the existing create-mode callers) are no longer sensitive to unquoted paths.

File: internal/image/overlay/session.go [hunk 3]
```diff
@@ -426,10 +446,11 @@ func (b *Builder) Postprocess(buildErr error) (err error) {
...
-	// parent of the emitted RAW artifact; emitOverlayArtifact has placed the .raw
-	// there, and the SBOM sidecar too when its best-effort copy succeeded (the
-	// summary lists whatever files are actually present, so a missing sidecar simply
-	// isn't shown).
+	// parent of the emitted artifact; emitOverlayArtifact has placed the .raw there
+	// (and the conversion above may have added qcow2/vhd/... alongside or in place of
+	// it), plus the SBOM sidecar when its best-effort copy succeeded. The summary
+	// lists whatever files are actually present, so it reflects the converted formats
+	// and a missing sidecar simply isn't shown.
 	display.PrintImageDirectorySummary(filepath.Dir(artifact), "RAW")
```
[the remark this answers is not in the document]
Comment (yktyagi) [reply internal/image/overlay/session.go]: Fixed. The summary's "Image Type" label is now derived from the template's `disk.artifacts` via a small `overlayArtifactTypeLabel` helper — it lists the requested output formats (e.g. "QCOW2", or "QCOW2, RAW") and defaults to "RAW" when no artifacts are declared. The artifact file list below the label already enumerates whatever files are actually present, so the box now reflects the converted formats rather than always claiming RAW.

File: internal/image/imageconvert/imageconvert.go [hunk 4]
```diff
@@ -287,7 +289,7 @@ func trimUnusedSpace(filePath string) error {
 	// Method 1: Try virt-sparsify if available (most effective)
 	if _, err := shell.ExecCmd("which virt-sparsify", false, shell.HostPath, nil); err == nil {
 		tempFile := filePath + ".sparse"
-		sparsifyCmd := fmt.Sprintf("virt-sparsify --in-place %s", filePath)
+		sparsifyCmd := fmt.Sprintf("virt-sparsify --in-place %s", shellSingleQuote(filePath))
```
[the remark this answers is not in the document]
Comment (yktyagi) [reply internal/image/imageconvert/imageconvert.go]: Good point — added `--` before the positional path for `virt-sparsify` as well, so a path beginning with `-` can't be misread as a flag. It now matches the qemu-img invocations (`virt-sparsify --in-place -- <quoted-path>`).

File: internal/image/imageconvert/imageconvert.go [hunk 5]
```diff
@@ -287,7 +289,7 @@ func trimUnusedSpace(filePath string) error {
 	// Method 1: Try virt-sparsify if available (most effective)
 	if _, err := shell.ExecCmd("which virt-sparsify", false, shell.HostPath, nil); err == nil {
 		tempFile := filePath + ".sparse"
-		sparsifyCmd := fmt.Sprintf("virt-sparsify --in-place %s", filePath)
+		sparsifyCmd := fmt.Sprintf("virt-sparsify --in-place -- %s", shellSingleQuote(filePath))
 		if _, err := shell.ExecCmd(sparsifyCmd, true, shell.HostPath, nil); err == nil {
```
[the remark this answers is not in the document]
Comment (yktyagi) [reply internal/image/imageconvert/imageconvert.go]: Good catch. Removed the `tempFile := filePath + ".sparse"` declaration and the `os.Remove(tempFile)` cleanup. With `--in-place`, virt-sparsify rewrites the image directly and produces no separate output file, so that cleanup created nothing to remove and risked deleting an unrelated pre-existing `<image>.sparse`. The failure path now just falls through to the qemu-img fallback.

File: internal/image/overlay/session.go [hunk 6]
```diff
@@ -426,15 +446,40 @@ func (b *Builder) Postprocess(buildErr error) (err error) {
...
+		if a.Type == "" {
+			continue
+		}
+		types = append(types, strings.ToUpper(a.Type))
+	}
+	if len(types) == 0 {
+		return "RAW"
+	}
+	return strings.Join(types, ", ")
+}
```
[the remark this answers is not in the document]
Comment (yktyagi) [reply internal/image/overlay/session.go:481]: Added `TestOverlayArtifactTypeLabel`, a table-driven test covering the edge cases: no/empty artifacts default to RAW, only-raw, a single type uppercased, multiple types joined in order, empty type entries skipped, and all-empty entries defaulting to RAW.

**Review by arodage — APPROVED** [review]
LGTM

**Review by srmungar — APPROVED** [review]
